src/main/java/org/spongepowered/common/mixin/command/vanilla/MixinCommandWorldBorder.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.command.vanilla;
+
+import net.minecraft.command.CommandWorldBorder;
+import net.minecraft.world.WorldServer;
+import net.minecraft.world.border.WorldBorder;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.common.SpongeImpl;
+
+@Mixin(CommandWorldBorder.class)
+public abstract class MixinCommandWorldBorder {
```

Comment:
The packet hack should be never applied when sending the packet. If `WorldProviderHell` is using a normal world border then the coordinates are already correctly in the overworld coordinate system which is how the Vanilla client will accept them.

The only thing that is not working correctly without a packet hack is the SpongeForge client. Normally, it should scale the coordinates to the nether system locally for nether worlds, however this is not happening currently because we also modify `WorldProviderHell` on the client.

So if the above is correct, the SF client must manually convert the world border center to the nether system and the server doesn't need to do anything additionally.